src/snowflake/snowpark/session.py

```diff
@@ -611,6 +616,27 @@ def __init__(
                 DEFAULT_COMPLEXITY_SCORE_UPPER_BOUND,
             ),
         )
+
+        self._thread_store = (
```

Comment:
I see quite many place is using this, maybe add a simple funciton in the utils.py for create_thread_store and create_lock to use it at different places

Reply:
added